R/generateHyperParsEffect.R

```diff
@@ -223,130 +239,157 @@ plotHyperParsEffect = function(hyperpars.effect.data, x = NULL, y = NULL,
   assertSubset(facet, choices = names(hyperpars.effect.data$data))
   assertFlag(pretty.names)
   assertFlag(global.only)
-  assert(checkClass(interpolate, "Learner"), checkString(interpolate),
+  assert(checkClass(interpolate, "Learner"), checkString(interpolate), 
          checkNull(interpolate))
   # assign learner for interpolation
-  if (checkClass(interpolate, "Learner") == TRUE ||
+  if (checkClass(interpolate, "Learner") == TRUE || 
       checkString(interpolate) == TRUE) {
     lrn = checkLearnerRegr(interpolate)
   }
   assertFlag(show.experiments)
   assertFunction(nested.agg)
-
+  
   if (length(x) > 1 || length(y) > 1 || length(z) > 1 || length(facet) > 1)
     stopf("Greater than 1 length x, y, z or facet not yet supported")
-
+  
   d = hyperpars.effect.data$data
   if (hyperpars.effect.data$nested)
     d$nested_cv_run = as.factor(d$nested_cv_run)
-
+  
   # set flags for building plots
   na.flag = any(is.na(d[, hyperpars.effect.data$measures]))
   z.flag = !is.null(z)
   facet.flag = !is.null(facet)
   heatcontour.flag = plot.type %in% c("heatmap", "contour")
-
-  # deal with NAs where optimizer failed
-  if (na.flag){
-    d$learner_status = ifelse(is.na(d[, "exec.time"]), "Failure", "Success")
-    for (col in hyperpars.effect.data$measures) {
-      col_name = stri_split_fixed(col, ".test.mean", omit_empty = TRUE)[[1]]
-      if (heatcontour.flag){
-        d[,col][is.na(d[,col])] = get(col_name)$worst
-      } else {
-        if (get(col_name)$minimize){
-          d[,col][is.na(d[,col])] = max(d[,col], na.rm = TRUE)
+  
+  # partial dep
+  # if heatcontour w k hypers, need interaction=T
+  # if iteration asked, need to use original hyperpars data
+  # if single hyper and single measure, pull from defaults
+  # if single hyper and multiple measures, 1 graph/multiple colored lines or many graphs/1 line
+  # if multiple hypers & single measure, pull from defaults and facet to get graph per hyper
+  # if multiple hypers & multiple measures, graph for each hyper/multiple colored lines or column for each measure
+  if ((hyperpars.effect.data$partial) && 
```

Comment:
yes basically.

i think you should match the resolution of the grid you use for interpolation with the size of the grid you use when calling partial dependence. i think you could unify the grid creation by using `generateFeatureGrid` which is currently in `generatePartialDependence.R` but should be moved to `utils.R`. It looks like this would serve your purposes without modification but if not I can make it more general so that it can. 

If you did that then you could call `doAggregatePartialDependence` directly which would have lower overhead. 

This would also allow you to remove points of extrapolation in the same way you are doing when not using partial dependence.

Also when plotting the points that were actually evaluated against the background which shows the interpolation/extrapolation, it might be good to color the points on the same scale as the background (you can do this by making the scales have the same name and range). Then you could see the difference between the performance value at the points and what the interpolation/extrapolation predicts them to be.

Otherwise looks really good!